#!/usr/bin/env python
PACKAGE = "drv_recognize"
from dynamic_reconfigure.parameter_generator_catkin import *

# name - a string which specifies the name under which this parameter should be stored
# type - defines the type of value stored, and can be any of int_t, double_t, str_t, or bool_t
# level - A bitmask which will later be passed to the dynamic reconfigure callback.
#         When the callback is called all of the level values for parameters that have been changed are ORed together and the resulting value is passed to the callback.
# description - string which describes the parameter
# default - specifies the default value
# min - specifies the min value (optional and does not apply to strings and bools)
# max - specifies the max value (optional and does not apply to strings and bools) 
gen = ParameterGenerator()
gen.add("hue_low_cfg", int_t, 0, "hue_low_cfg", 26, 0, 255)
gen.add("sat_low_cfg", int_t, 0, "sat_low_cfg", 34, 0, 255)
gen.add("val_low_cfg", int_t, 0, "val_low_cfg", 46, 0, 255)
gen.add("hue_high_cfg", int_t, 0, "hue_high_cfg", 77, 0, 255)
gen.add("sat_high_cfg", int_t, 0, "sat_high_cfg", 255, 0, 255)
gen.add("val_high_cfg", int_t, 0, "val_high_cfg", 255, 0, 255)

exit(gen.generate(PACKAGE, "drv_recognize", "recog"))